UPGRADE_GUIDE.md

```diff
@@ -2,22 +2,130 @@
+        404 status code)
+  - `EasyPost.Exception.PaymentError` - thrown when a payment error occurs (402 status code)
+  - `EasyPost.Exception.ProxyError` - thrown when the library is unable to connect to the API via a proxy
+  - `EasyPost.Exception.RateLimitError` - thrown when the API rate limit is exceeded (429 status code)
+  - `EasyPost.Exception.RedirectError` - thrown when the http status is between 300 and 308.
+  - `EasyPost.Exception.ServiceUnavailableError` - thrown when the API is unavailable (503 status code)
+  - `EasyPost.Exception.TimeoutError` - thrown when the API request times out (408 status code)
+  - `EasyPost.Exception.UnauthorizedError` - thrown when the API receives an unauthorized request (401 or 403 status
+        code)
+  - `EasyPost.Exception.UnknownApiError` - thrown when an unknown API error occurs (unexpected 4xx status code)
+- `EasyPost.Exception.General` for Generic error returned by the client library. Subclasses of this exception type are:
+  - `EasyPost.Exceptions.FilteringError` - thrown when an error occurs during filtering (e.g. calculating lowest rate)
+  - `EasyPost.Exceptions.InvalidObjectError` - thrown when an invalid object is being used
+  - `EasyPost.Exceptions.InvalidParameterError` - thrown when an invalid parameter is being used
+  - `EasyPost.Exceptions.MissingPropertyError` - thrown when a required property is missing (e.g. `Id` for most objects)
+  - `EasyPost.Exceptions.SignatureVerificationError` - thrown when the signature for a webhook is invalid
+
+Any exception thrown by the EasyPost library will be one of the above types.
+
+Any `EasyPost.Exception.API` exception will have the following properties:
+
+- `String Code` - the HTTP status code returned by the API call (e.g. 404)
+- `String Message` - the error message returned by the API (e.g. "PARAMETER.INVALID")
+- `List<Error> Errors` - a list of errors returned by the API (e.g. "Invalid parameter: to_address")
+
+Users can better anticipate exception information through utilities in the `EasyPost.Exception.Constants` file.
+
+## 6.0 Uniform empty response
+
+*Likelihood of Impact: **High***
+
+Below functions return empty bodies or boolean have been changed to void.
+
+- `fundWallet()` and `deletePaymentMethod()` in Billing class
+- `createList()` in Tracker class
+- `updateEmail()` in ReferralCustomer class
+
+## 6.0 Services and Resources
+
+*Likelihood of Impact: **Low***
+
+Static and instance-based methods have been divided into separate services and resources, methods that have API interactions are in services now. For example, the static method `Shipment.Create` is now accessible in the Shipment service via `client.Shipment.Create`. The instance method `myShipment.lowestRate` is still accessible only via a Shipment instance.
+
+Instances of an object cannot be initialized directly. Instead, use the service to create the instance via a `Create`, `Retrieve`, or `All` API call.
```

Comment:
I'd argue that the likelyhood of this change is `high` because there are many function calls that are no longer instance-based. Please bump this up in the list and provide an example of something that used to be instance but is now static. eg: buying a shipment is no longer `shipment.buy()` but is now `client.shipment.buy("shp_...")`

Reply:
Resolved and splited the sentence to three bullet points